photutils/psf/models.py

```diff
@@ -12,8 +12,274 @@
+    particular spline interpolator does not support weighted smoothing,
+    :py:class:`Discrete2DModel` can be used to store image weights
+    so that these can be passed to the fitter.
+
+    The fittable model provided by this class has three model parameters:
+    total flux of the underlying image, and two shifts along each axis of the
+    image.
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        Array containing 2D image.
+
+    origin : tuple, None, optional
+        Origin of the coordinate system in image pixels. Origin indicates where
+        in the image model coordinates ``x`` and ``y`` are zero. If `origin` is
+        `None`, then model's origin will be set to the center of the image.
+
+    weights : numpy.ndarray, None, optional
+        An array of weights for the corresponding data. [Currently not used]
+
+    fillval : float, optional
+        The value to be returned by the `evaluate` or
+        ``astropy.modeling.Model.__call__`` methods
+        when evaluation is performed outside the definition domain of the
+        model.
+
+    kwargs : dict, optional
+
+        Additional optional keyword arguments to be passed directly to the
+        `compute_interpolator` method. Possible values are:
+
+        - **degree** : int, tuple, optional
+            Degree of the interpolating spline. A tuple can be used to provide
+            different degrees for the X- and Y-axes. Default value is degree=3.
+
+        - **s** : float, optional
+            Non-negative smoothing factor. Default value s=0 corresponds to
+            interpolation.
+            See :py:class:`~scipy.interpolate.RectBivariateSpline` for more
+            details.
+
+    """
+    flux = Parameter(description='Total flux of the image.', default=None)
+    x_0 = Parameter(description='Shift along the X-axis relative to the '
+                    'origin.', default=0.0)
+    y_0 = Parameter(description='Shift along the Y-axis relative to the '
+                    'origin.', default=0.0)
+
+    def __init__(self, data, flux=flux.default,
+                 x_0=x_0.default, y_0=y_0.default,
+                 origin=None, weights=None, fillval=0.0, kwargs={}):
+        """
+        """
+        self._fillval = fillval
+        self._weights = weights
+
+        # compute flux and normalize data so that sum(data) = 1:
+        tflux = np.sum(data, dtype=np.float64)
+        if tflux == 0.0 or not np.isfinite(tflux):
+            tflux = 1.0
+        if flux is None:
+            flux = tflux
+        self._ndata = data / tflux
+
+        # set input image related parameters:
+        self._ny, self._nx = data.shape
+
+        # find origin of the coordinate system in image's pixel grid:
+        self.origin = origin
+
+        super(Discrete2DModel, self).__init__(flux, x_0, y_0)
+
+        # define interpolating spline:
+        self.compute_interpolator(kwargs)
+
+    @property
+    def ndata(self):
```

Comment:
how about `normalized_data`?  `ndata` sounds like it could be `nddata` or number of data points or something like that.